src/Common/OvercommitTracker.cpp

```diff
@@ -75,10 +80,14 @@ OvercommitResult OvercommitTracker::needToStopQuery(MemoryTracker * tracker, Int
 
     required_memory += amount;
     auto wait_start_time = std::chrono::system_clock::now();
-    bool timeout = !cv.wait_for(lk, max_wait_time, [this, id]()
+    bool timeout;
     {
-        return id < id_to_release || cancellation_state == QueryCancellationState::NONE;
-    });
+        CurrentMetrics::Increment metric_increment(CurrentMetrics::ThreadsInOvercommitTracker);
```

Comment:
I would move it to the beginning of the function, so it will count all threads (including, for example, threads waiting for `global_lock` or `overcommit_m`)